developerportal/apps/common/utils.py

```diff
@@ -21,12 +23,14 @@ def _resolve_model(model):
 def _prep_search_terms(terms: str) -> str:
     """Fix up provided terms ready for passing to Wagtail's .search() model
     manager method. Note that at a lower level, for safety, the postgres_search
-    backend does escaping and quoting for us: See Lexeme().as_sql() in
+    backend ALSO does escaping and quoting for us: See Lexeme().as_sql() in
     https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/postgres_search/query.py  # noqa
     """
+
     terms = unquote(terms)
     NON_SPACE_WHITESPACE = "\n\t\r"
     terms = terms.translate(str.maketrans("", "", NON_SPACE_WHITESPACE))
+    terms = bleach.clean(terms)
```

Comment:
This may seem a bit surprising, to be bleaching on the way in but it serves two purposes

1) I feel better about treating all input as dangerous, even though Wagtail Search also escapes at a lower level

2) Testing locally, with the string `<marquee>` in some test content, I tried searching and - before bleaching to escape the content - a search for "<marquee>" would return no results, even though it should have found the page with the test content. When it the input was bleached, however, it WAS found. (My assumption here is that the sanitised/escaped input better matches the string in the wagtailsearch index, and talking with one of the core team it appears this could well be the case, and is a bug that they will address)

Reply:
Yeah, I thikn I'll add a test that fails and prints a reminder message when the Wagtail version increases - see https://github.com/wagtail/wagtail/issues/6131

- [ ] add canary test